apps/web-client/src/routes/orders/suppliers/reconcile/[id]/+page.svelte

```diff
@@ -7,95 +7,55 @@
-					ordered: 3
-				},
-				{
-					isbn: "9780987654314",
-					title: "Thinking, Fast and Slow",
-					authors: "Daniel Kahneman",
-					price: 12.99,
-					delivered: false,
-					ordered: 4
-				}
-			]
-		}
-	];
-
-	const reconciliation = {
-		id: 123,
-		lastUpdated: new Date()
-	};
-
-	let isbn = "";
-	let books: Array<{
-		isbn: string;
-		title: string;
-		authors: string;
-		price: number;
-		quantity: number;
-	}> = [];
-
-	// Mock supplier orders data
-	const selectedOrders = [
-		{ id: 1, supplier: "Academic Books Ltd", books: 5 },
-		{ id: 2, supplier: "Penguin Random House", books: 3 }
-	];
-
-	function handleIsbnSubmit() {
+	import type { PageData } from "./$types";
+	import { addOrderLinesToReconciliationOrder, finalizeReconciliationOrder } from "$lib/db/cr-sqlite/order-reconciliation";
+	import { page } from "$app/stores";
+	import { onDestroy, onMount } from "svelte";
+	import { invalidate } from "$app/navigation";
+	import { processOrderDelivery } from "$lib/db/cr-sqlite/utils";
+	import { defaults, superForm } from "sveltekit-superforms";
+	import { zod } from "sveltekit-superforms/adapters";
+	import { scannerSchema, bookSchema } from "$lib/forms/schemas";
+
+	// implement order reactivity/sync
+	export let data: PageData;
+
+	// #region reactivity
+	let disposer: () => void;
+	onMount(() => {
+		// NOTE: ordersDbCtx should always be defined on client
+		const { rx } = data;
+
+		const disposer1 = rx.onPoint("reconciliationOrder", BigInt($page.params.id), () => invalidate("reconciliationOrder:data"));
+		const disposer2 = rx.onRange(["reconciliation_order", "reconciliation_order_lines"], () => invalidate("reconciliationOrder:data"));
+		disposer = () => (disposer1(), disposer2());
+	});
+
+	//#endregion reactivity
+
+	onDestroy(async () => {
+		// Unsubscribe on unmount
+		disposer();
+	});
+	$: books = data?.reconciliationOrderLines || [];
+
+	async function handleIsbnSubmit(isbn: string) {
 		if (!isbn) return;
 
-		// Mock adding a book
-		books = [
-			{
-				isbn,
-				title: "Sample Book",
-				authors: "Sample Author",
-				price: 19.99,
-				quantity: 1
-			},
-			...books
-		];
-		isbn = "";
+		await addOrderLinesToReconciliationOrder(data.ordersDb, parseInt($page.params.id), [{ isbn, quantity: 1 }]);
 	}
 
-	$: totalDelivered = mockSupplierBooks.reduce((acc, supplier) => acc + supplier.books.filter((b) => b.delivered).length, 0);
-	$: totalOrdered = mockSupplierBooks.reduce((acc, supplier) => acc + supplier.books.length, 0);
+	const form = superForm(defaults(zod(scannerSchema)), {
+		validators: zod(scannerSchema),
+		validationMethod: "submit-only"
+	});
+
+	const { form: formStore } = form;
+
+	$: placedOrderLines = data?.placedOrderLines;
+	$: totalDelivered = data?.reconciliationOrderLines.map((book) => book.quantity).reduce((acc, curr) => acc + curr, 0);
+	$: totalOrdered = placedOrderLines.length;
+	$: processedOrderDelivery = processOrderDelivery(data?.reconciliationOrderLines, data?.placedOrderLines);
```

Comment:
This will re-run ever time we add new lines. This is a downside of having all of this logic packed in one view. A slight improvement would be to move it into the if block where the ComparisonTable is rendered as a `{@const processedOrderDeliver = ...}` declaration